PlayList.java

```diff
@@ -35,32 +35,59 @@ public Track getTrack(int index) {
      *  If the list is full, does nothing and returns false.
      *  Otherwise, appends the track and returns true. */
     public boolean add(Track track) {
-        //// replace the following statement with your code
+        int size1 = this.getSize();
+        if(size1 == this.getMaxSize()){
+            return false;
+        }
+        else{
+            tracks[size1] = track;
+            size = size1 + 1;
+        }
         return true;
     }
 
     /** Returns the data of this list, as a string. Each track appears in a separate line. */
     //// For an efficient implementation, use StringBuilder.
     public String toString() {
-        //// replace the following statement with your code
-        return "";
+        StringBuilder trackString = new StringBuilder();
+        for (int i = 0 ; i < this.getSize(); i++){
+            trackString.append("\n");
+            trackString.append(tracks[i].toString());
+        }
+        return trackString.toString();
     }
 
     /** Removes the last track from this list. If the list is empty, does nothing. */
      public void removeLast() {
-        //// replace this comment with your code
+        if(this.getSize() > 0 ){
+            tracks[size - 1] = null;
+            size = this.getSize() - 1;
+        }
```

Comment:
Internally the object, we use properties directly - getters mostly use external the object.
```suggestion
        if(this.size > 0 ){
            tracks[this.size - 1] = null;
            size--;
        }
```